### PR TITLE
Debug and try re-rendering for Python 3.12

### LINKED ISSUE #3703: PyBaMM v23.9 hasn't made it to conda-forge yet
I suppose this is because we were waiting to add support for Python 3.12 here first, which is complete now – there is a stale PR at https://github.com/conda-forge/pybamm-feedstock/pull/53, which is probably why a 23.9 build did not get triggered at that time.

This issue is being opened for purposes of visibility – we should ensure that this is resolved before 24.1 (non-RC) is released on PyPI.

### PR DESCRIPTION
**Pull request by agriyakhetarpal**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Closes pybamm-team/PyBaMM#3703

I am trying to re-render for 23.9 through this PR from my fork. Bumped up the build number so that it can be reset back to zero afterwards when the version is bumped.

### CODE REVIEW

**Comment by agriyakhetarpal** [comment]
@conda-forge-admin, please rerender

**Comment by agriyakhetarpal** [comment]
Unfortunately we can't see the logs from #54, probably because it has been two months and their retention period is over. That PR was automated and _should_ have kicked in the 23.9 upload on conda-forge in an ideal situation, but we just have 23.5 here: https://anaconda.org/conda-forge/pybamm

I have verified the logs here and it builds PyBaMM 23.9, hopefully this resolves the issue and brings it to conda-forge. I will merge after approving comments from @priyanshuone6 and @Saransh-cpp – please check if this looks right!

<hr>

N.B. #53 was closed because it became outdated (as mentioned in the thread there). We should have merged it before merging #54. Once we get to the release of 24.1 next month, we can update the Python version selectors here to include Python 3.8–3.12, which should kick off a re-render (or we can do that manually)

**Review by priyanshuone6 — APPROVED** [review]

**Comment by actualpriyanshu** [comment]
Thanks, I noticed v23.9 is now available

**Pull request merged**
